src/httpserver/handlers/handler.clj

```diff
@@ -1,24 +1,27 @@
 (ns httpserver.handlers.handler
-  (:require [httpserver.router :as router]))
+  (:require [httpserver.routing.router :as router]))
 
 (import '(com.td.HttpServer HttpResponse))
 (import '(com.td.HttpServer IHandler))
 (import '(com.td.HttpServer HttpRequest))
 
 (defn build-request [httpRequest]
-  (hash-map :path (.path httpRequest) :method (.method httpRequest) :headers (.headers httpRequest) :body (.body httpRequest)))
+  {:path (.path httpRequest) :method (.method httpRequest) :headers (.headers httpRequest) :body (.body httpRequest)})
+
+(def routes {})
 
 (deftype main-handler []
   com.td.HttpServer.IHandler
   (generateResponse [this httpRequest]
     (let [request (build-request httpRequest)
-          function-to-use (router/get-function request)
+          function-to-use (router/get-handler request routes)
           response (function-to-use request)
           headers (java.util.HashMap. (:headers response))]
       (doto (HttpResponse.)
         (.setBody (:body response))
         (.setResponseCode (:code response))
         (.addHeaders headers)))))
 
-(defn new-handler []
+(defn new-handler [list-of-routes]
+  (def routes list-of-routes)
```

Comment:
If you want to use state, rather than re-`def`-ing, you should use an "atom". However, I don't think you need to do that here. I **think** you could do something like this:

```clojure
(deftype main-handler [routes]
  com.td.HttpServer.IHandler
  (generateResponse [this httpRequest]
    ;; pretty sure you have access to routes here, since the handler was constructed with it
    ...))

(defn new-handler [list-of-routes]
  (main-handler. list-of-routes))
```

Now you're back to being stateless! Hooray!